### PR TITLE
Ds 371 contracted hats the sequel

### PR DESCRIPTION
**Pull request by dataswifty**

### CODE REVIEW

File: hat/app/org/hatdex/hat/api/controllers/RichData.scala [hunk 1]
```diff
@@ -95,7 +83,7 @@ class RichData @Inject() (
   val adjudicatorScheme =
     configuration.underlying.getString("adjudicator.scheme")
   val adjudicatorEndpoint = s"${adjudicatorScheme}${adjudicatorAddress}"
-  val adjudicatorClient = new NetworkRequest(adjudicatorEndpoint, wsClient)
+  val adjudicatorClient = new AdjudicatorRequest(adjudicatorEndpoint, wsClient)
```
Comment (augustinas) [inline hat/app/org/hatdex/hat/api/controllers/RichData.scala:86]: Much better name

File: hat/app/org/hatdex/hat/api/service/applications/ApplicationsService.scala [hunk 2]
```diff
@@ -158,17 +161,18 @@ class ApplicationsService @Inject() (
...
         adjudicatorClient
           .joinContract(
             hatName,
-            io.dataswift.adjudicator.Types
-              .ContractId(
-                java.util.UUID.fromString(application.id)))
-      case _ => Future.successful(Unit)
+            ContractId(UUID.fromString(application.id)))
+      case _ => {
+        Future.successful(Unit)
```
Comment (augustinas) [inline hat/app/org/hatdex/hat/api/service/applications/ApplicationsService.scala:172]: Suggest replacing `Unit` with `akka.Done`. Successful Future that returns a Unit can lead to unexpected results, especially when flattening Futures.

File: hat/conf/application.conf [hunk 3]
```diff
@@ -234,7 +234,7 @@ exchange {
 
 
 adjudicator {
-  address = "adjudicator.hubofallthings.com"
+  address = "contracts.dataswift.dev"
   adresss = ${?ADJUDICATOR_ADDRESS}
   scheme = "http://"
```
Comment (augustinas) [inline hat/conf/application.conf:239]: Should it use `https` schema?

File: hat/test/org/hatdex/hat/api/service/applications/ApplicationsServiceSpec.scala [hunk 4]
```diff
@@ -470,6 +465,47 @@ class ApplicationsServiceSpec(implicit ee: ExecutionEnv)
...
+        println(contractApp)
+        println(notablesApp)
+        notablesApp shouldEqual (Unit)
+        //contractApp must beLeft(ServiceRespondedWithFailure("The Adjudicator Service responded with an error: Internal Server Error"))
+      }
+
+      result await (1, 20.seconds)
+    }
+
+    // Commented until the infrastructure can handle it.
```
Comment (augustinas) [inline hat/test/org/hatdex/hat/api/service/applications/ApplicationsServiceSpec.scala:485]: Looks like this behaviour should be mocked?

File: hat/app/org/hatdex/hat/api/controllers/RichData.scala [hunk 5]
```diff
@@ -664,24 +654,30 @@ class RichData @Inject() (
...
+      {
+        publicKeyReqsponse match {
+          case Left(PublicKeyRequestFailure.ServiceRespondedWithFailure(failureDescription)) => {
+            Left(ContractVerificationFailure.ServiceRespondedWithFailure(s"The Adjudicator Service responded with an error: ${failureDescription}"))
+          }
+          case Left(PublicKeyRequestFailure.InvalidPublicKeyFailure(failureDescription)) => {
+            Left(ContractVerificationFailure.ServiceRespondedWithFailure(s"The Adjudicator Service responded with an error: ${failureDescription}"))
+          }
+          case Right(PublicKeyReceived(publicKey)) => {
             val tryJwtClaim = ShortLivedTokenOps.verifyToken(
```
Comment (benshaw) [inline hat/app/org/hatdex/hat/api/controllers/RichData.scala:669]: i find nested case statements hard to read but potentially its a personal thing
in this case you could extract the logic into a method and pass the publicKey
give the method a good name and i know what it does i dont have to dig through the details

File: hat/app/org/hatdex/hat/utils/AdjudicatorRequest.scala [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+object PublicKeyRequestFailure {
+  final case class ServiceRespondedWithFailure(failureDescription: String)
+    extends PublicKeyRequestFailure
+  final case class InvalidPublicKeyFailure(failureDescription: String)
+    extends PublicKeyRequestFailure
+}
+
+sealed trait PublicKeyRequestSuccess
+object PublicKeyRequestSuccess {
+  final case class PublicKeyReceived(publicKeyAsByteArray: Array[Byte])
```
[the remark this answers is not in the document]
Comment (dataswifty) [reply hat/app/org/hatdex/hat/utils/AdjudicatorRequest.scala:24]: Reasonable.

**Pull request merged**